budgetportal/models.py

```diff
@@ -1809,6 +1808,67 @@ def _build_complete_expenditure(cls, records):
         return complete_expenditure
 
 
+class Language(models.Model):
+    name = models.CharField(max_length=255)
+
+    def __str__(self):
+        return self.name
+
+
+prov_keys = prov_abbrev.keys()
+prov_choices = tuple([(prov_key, prov_key) for prov_key in prov_keys])
+
+
+class Event(models.Model):
+    date = models.CharField(max_length=255)
+    type = models.CharField(max_length=255, choices=(
+        ('hackathon', 'hackathon'), ('dataquest', 'dataquest'), ('cid', 'cid'),
+        ('gift-dataquest', 'gift-dataquest'),
+    ))
+    province = models.CharField(max_length=255, choices=prov_choices)
+    where = models.CharField(max_length=255)
+    url = models.URLField(blank=True, null=True)
+    notes_url = models.URLField(blank=True, null=True)
+    video_url = models.URLField(blank=True, null=True)
+    rsvp_url = models.URLField(blank=True, null=True)
+    presentation_url = models.URLField(blank=True, null=True)
+    status = models.CharField(max_length=255, default='upcoming', choices=(('upcoming', 'upcoming'), ('past', 'past')))
+
+    def __str__(self):
+        return "{} {} ({} {})".format(self.type, self.date, self.where, self.province)
+
+class Video(models.Model):
```

Comment:
mmmm a much simpler model would be

```
Video  1---------------------*  VideoLanguage
- title_id                        - label
- title                           - youtube_id
- description
```
then you wouldn't need custom grouping with a string ID and duplication of content

Reply:
The languages per video need to be ordered as before.
I'm happy for us to pick the easiest possible way of getting that done quickly. Either by adding an integer field like `sort_position` and sorting on that, or using a library for it - this looks like the best library by features, documentation, and maintenance https://github.com/alsoicode/django-admin-sortable

Reply:
I plan on fixing this in step 3. Will link to and from this comment when completed.

Reply:
This is now implemented in https://github.com/vulekamali/datamanager/pull/161